### ISSUE TITLE
(suggestion) dont tp player to spawn when "fancy respawn" is off

### ISSUE THREAD

**Issue by muffinized** [cervinakuy/KitPvP#79]
i need this option because i have a plugin that shows the death animation upon death, but becausethis plugin teleports you to spawn instantly, the animation plays at spawn. so can u make it so the plugin does NOT do ANY movement or teleportation to the player at death when fancy respawn is off cuz my plugin is affected by it

**Comment by cervinakuy** [comment]
Then where is the player supposed to be teleported when they respawn? At their death point? What?

**Comment by muffinized** [comment]
no, let /setworldspawn thing respawn the player. i have a plugin which shows the death animation. and when ur plugin is tping the player to spawn, the animation plays at spawn, which is bad. so the best way is to not enable the deathlistener if fancy respawn is off.

**Comment by muffinized** [comment]
@cervinakuy

**Comment by cervinakuy** [comment]
Hmm... maybe not after the next update (I think I made a change that should allow it to work). Let me know if it happens in 2.1.6 and I will look into adding a solution.

**Comment by muffinized** [comment]
2.1.6 aint released

**Comment by MKieburtz** [comment]
@cervinakuy 

I would also like an option to not teleport the player after death, and let the respawning be handled by another plugin. On my arena server the player is automatically respawned by the multiverse core plugin, and then a second later KitPVP teleports them to the arena spawn. There is also a large lag spike associated with this second teleport, but I don't know why it happens. 

Frankly I don't know why more people aren't complaining about this.  The setup of multiple worlds using multiverse with multiple arenas using KitPVP can't be that uncommon of a use case. 

I'm a developer and would be happy to open a PR for this if you approve the idea.

**Comment by cervinakuy** [comment]
@MKieburtz What is your setup? Do you have FancyDeath enabled?

I think the most feasible option would be to remove [these lines](https://github.com/cervinakuy/KitPvP/blob/879693a66d10d92994ebd40e8a4b16315cbdd933/src/main/java/com/planetgallium/kitpvp/listener/DeathListener.java#L73-L95) altogether, and I think it would accomplish what the OP desired along with your use case.

> There is also a large lag spike associated with this second teleport, but I don't know why it happens.
Are you on the latest version, or an older version? In older versions, updating the stats (which happens at death) would cause lag spikes. If you are on the latest version, this would still dependent on whether you have FancyDeath enabled or not.

**Comment by MKieburtz** [comment]
I do not have fancy death enabled, that makes it worse actually. With fancy death enabled the player put into spectator and respawned at multiverse's respawn point, then teleported back to their death spot for a couple seconds, and finally teleported back to the kitpvp arena spawn point. It's very jarring.

I am a couple versions behind. Currently on 2.1.7. I haven't updated because I saw that MySQL support was disabled and my arena is still on 1.16.5. If I jumped to the newest version would stats stored in MySQL be transferred to SQLite?
